### ISSUE TITLE
With NET 9 RC2 SDK installed, running projects created with CLI result in unwanted strings on VS winserver2012 R2

### ISSUE THREAD

**Issue by v-pingpchen** [dotnet/msbuild#10841]
### Issue Description

Build info:
9.0.100-rc.2.24428.4 (runtime-9.0.0-rc.2.24426.3)

### Steps to Reproduce

1. Download NET 9 RC2 SDK on win-x64
2. Create a console project with CLI
    dotnet new console
3. Build the console
    dotnet build
4. Run the console
    dotnet run

### Expected Behavior

Project runs successfully without any unwanted strings

### Actual Behavior

Unwanted strings are shown together with the output
![Image](https://github.com/user-attachments/assets/ae8b5286-fafc-429b-8887-4411f133c3f5)

### Versions & Configurations

dotnet --info
.NET SDK:
Version: 9.0.100-rc.2.24428.4
Commit: 0030622fd9
Workload version: 9.0.100-manifests.12008748
MSBuild version: 17.12.0-preview-24426-01+14df0589a

Runtime Environment:
OS Name: Windows
OS Version: 10.0.20348
OS Platform: Windows
RID: win-x64
Base Path: C:\Program Files\dotnet\sdk\9.0.100-rc.2.24428.4\

.NET workloads installed:
Configured to use loose manifests when installing new manifests.
There are no installed workloads to display.

Host:
Version: 9.0.0-rc.2.24426.3
Architecture: x64
Commit: static

.NET SDKs installed:
9.0.100-rc.2.24428.4 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
Microsoft.AspNetCore.App 6.0.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 9.0.0-rc.2.24426.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 9.0.0-rc.2.24426.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 9.0.0-rc.2.24425.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
x86 [C:\Program Files (x86)\dotnet]
registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
Not set

global.json file:
Not found

Learn more:
https://aka.ms/dotnet/info

Download .NET:
https://aka.ms/dotnet/download

**Comment by MichalPavlik** [comment]
Relates to https://github.com/dotnet/msbuild/issues/10579

@baronfel, I can enhance my previous fix to cover this case (using old Windows version without VT100 support). Do you think the change will be accepted for .NET 9 GA?

**Comment by baronfel** [comment]
No, we are now frozen for GA. We could take it for the first servicing release though.

**Comment by MichalPavlik** [comment]
Fixed in 9.0.101

**Comment by vdanche** [comment]
Fixed on 9.0.101
![Image](https://github.com/user-attachments/assets/045fea32-4ef4-4f5c-9119-1499a773759c)
